flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/calcite/FlinkCalciteSqlValidator.java

```diff
@@ -125,4 +126,17 @@ public void validateColumnListParams(
         // this makes it possible to ignore them in the validator and fall back to regular row types
         // see also SqlFunction#deriveType
     }
+
+    @Override
+    public void validateInsert(SqlInsert insert) {
+        // TODO Seems like this validation breaks the SqlToOperationConverter,
```

Comment:
I think it would be better if we created an issue for it and then linked to it. :-) I would open it as an improvement: it'd be a good thing to run the validation, but of course we need to describe what's stopping us from it (including one of the examples that break).